numpy/lib/function_base.py

```diff
@@ -3191,11 +3184,15 @@ def i0(x):
     --------
     >>> np.i0(0.)
     array(1.0)  # may vary
-    >>> np.i0([0., 1. + 2j])
-    array([ 1.00000000+0.j        ,  0.18785373+0.64616944j])  # may vary
+    >>> np.i0([0, 1, 2, 3])
+    array([1.        , 1.26606588, 2.2795853 , 4.88079259])  # may vary
 
     """
     x = np.asanyarray(x)
+    if x.dtype.kind == 'c':
+        raise ValueError("i0 requires real input")
```

Comment:
```suggestion
        raise TypeError("i0 not supported on complex values")
```
makes it a little more clear